src/utils.py

```diff
@@ -157,6 +161,25 @@ def partial_match(pattern, actual):
 #
 #
 
+def call_n_times(fn, protect_from, num_attempts=3):
+    "if after calling `num_attempts` it fails to return a value, it will return None"
```

Comment:
`None` does not seem a particularly informative value in case of prolonged failures, but at least whatever is calling will fail too e.g. when it tries to access a field (and we have longs)